cartography/intel/gcp/storage.py

```diff
@@ -0,0 +1,223 @@
+                ("You do not have storage.bucket.get access to the bucket %s. "
+                 "Full details: %s"), bucket, e, )
+            return None
+        else:
+            raise
+
+
+def get_gcp_buckets(storage, project):
+    """
+    Returns a list of storage objects within some given project
+
+    :type storage: A storage resource object
+    :param storage: The storage resource object created by googleapiclient.discovery.build()
+
+    :type project: str
+    :param project: The Google Project name that you are retrieving buckets from
+    :rtype: Storage Object
+    :return: Storage response object
+    """
+    try:
+        req = storage.buckets().list(project=project)
+        res = req.execute()
+        return res
+    except HttpError as e:
+        reason = compute._get_error_reason(e)
+        if reason == 'invalid':
+            logger.debug(
+                (
+                    "The project %s is invalid - returned a 400 invalid error."
+                    "Full details: %s"
+                ),
+                project,
+                e,
+            )
+            return None
+        elif reason == 'forbidden':
+            logger.debug(
+                ("You do not have storage.bucket.list access to the project %s. "
+                 "Full details: %s"), project, e, )
+            return None
+        else:
+            raise
+
+
+def get_gcp_bucket_iam_policy(storage, bucket):
+    """
+    Retrieves IAM policy about the given bucket.
+
+    :type storage: A storage resource object
+    :param storage: The storage resource object created by googleapiclient.discovery.build()
+
+    :type bucket: str
+    :param bucket: Google Cloud Bucket name
+
+    :rtype: dict
+    :return: IAM Policy for specified bucket
+    """
+    try:
+        req = storage.buckets().getIamPolicy(bucket=bucket)
+        res = req.execute()
+        return res
+    except HttpError as e:
+        reason = compute._get_error_reason(e)
+        if reason == 'notFound':
+            logger.debug(
+                ("The bucket %s was not found - returned a 404 not found error."
+                 "Full details: %s"), bucket, e, )
+            return None
+        elif reason == 'forbidden':
+            logger.debug(
+                ("You do not have storage.bucket.getIamPolicy access to the bucket %s. "
+                 "Full details: %s"), bucket, e, )
+            return None
+        else:
+            raise
+
+
+def transform_gcp_buckets(bucket_res): 
+    '''
+    Transform the GCP Storage Bucket response object for Neo4j ingestion
+    
+    :type bucket_res: A storage resource object (https://cloud.google.com/storage/docs/json_api/v1/buckets) 
+    :param bucket_res: The return data
+    
+    :rtype: list
+    :return: List of buckets ready for ingestion to Neo4j
+    '''
+    
+    bucket_list = []
+    for b in bucket_res.get('items', []):
+        bucket = {}
+        bucket['etag'] = b.get('etag', '')  
```

Comment:
I think we should replace all of the `dict.get('value', '')` calls with `dict.get('value')`.  I'd prefer not to have a field as an empty string ingested to the graph.

Reply:
Good call, done.